test/exercise/arrays/solution.rb

```diff
@@ -1,12 +1,38 @@
 module Exercise
   module Arrays
     class << self
+      def max(array)
+        array.reduce { |acc, current | acc > current ? acc : current }
+      end
+
       def replace(array)
-        array
+        maximum = max(array)
+        array.map { |number| number > 0 ? maximum : number }
       end
 
-      def search(_array, _query)
-        0
+      def search(_array, _query, index = 0)
+        length = _array.length
+
+        if length == 0
+          return -1
+        end
+
+        if length == 1
+          return _array[0] == _query ? index : -1
+        end
+
+        left_part, right_part = _array.each_slice((_array.size/2.0).round).to_a
```

Comment:
`(_array.size/2.0).round` -- I believe, this can be replaced with just `(_array.size/2)`.

Reply:
```
irb(main):007:0> [1, 2, 3, 4, 5].each_slice(5 / 2).to_a
=> [[1, 2], [3, 4], [5]]
irb(main):008:0> [1, 2, 3, 4, 5].each_slice((5 / 2.0).round).to_a
=> [[1, 2, 3], [4, 5]]
```